pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -435,28 +441,31 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 		if remaining < need {
 			var errStr string
 			var entities []*types.ErrorEntity
-			errSeverity := types.ErrorSeverityWarning
-			retryCondition := "Retry will be triggered when apps will shutdown: "
+			var domains []string
+			errSeverity := types.ErrorSeverityError
 			if remaining+halting < need {
 				errStr = fmt.Sprintf("Remaining memory bytes %d app instance needs %d",
 					remaining, need)
-				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
-					status := st.(types.AppInstanceStatus)
-					entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
-					retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
-				}
 			} else {
-				errStr = fmt.Sprintf("App instance needs %d bytes but only have %d; waiting for halting app instances to free up %d bytes",
+				errStr = fmt.Sprintf("App instance needs %d bytes but only have %d; waiting for one or more halting app instances to free up %d bytes",
 					need, remaining, halting)
 				errSeverity = types.ErrorSeverityNotice
-				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
-					status := st.(types.AppInstanceStatus)
-					if status.Activated || status.ActivateInprogress {
-						entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
-						retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
-					}
+			}
+			for _, st := range ctx.pubAppInstanceStatus.GetAll() {
+				status := st.(types.AppInstanceStatus)
+				if status.Activated || status.ActivateInprogress {
+					entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
```

Comment:
Doesn't this still include the app instance which is trying to boot? That was the error Sahil pointed out in the demo.

Reply:
We have such logic in getRemainingMemory and also I can see that we mark domain as ActivateInprogress and Activate after successfully checking of memory needed. The error showed before was about lack of check for `status.Activated || status.ActivateInprogress` in `remaining+halting < need` branch I think.

Reply:
ok